server/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java

```diff
@@ -99,10 +97,10 @@ protected void documentation(StringBuilder sb) {
     public RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
         final String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
         final IndicesOptions indicesOptions = IndicesOptions.fromRequest(request, IndicesOptions.strictExpand());
-        if (request.hasParam("local")) {
-            DEPRECATION_LOGGER.deprecate("local", LOCAL_DEPRECATED_MESSAGE);
+        // needed only in v8 to catch breaking usages and may be removed in v9
+        if (request.hasParam("local")  && Version.CURRENT.major == Version.V_7_0_0.major + 1) {
+            throw new IllegalArgumentException("parameter [local] is not supported");
```

Comment:
This is unnecessary, as mentioned in https://github.com/elastic/elasticsearch/pull/64867#discussion_r520703765